src/components/Chart/Chart.js

```diff
@@ -0,0 +1,297 @@
+import React, { Component, Fragment } from 'react'; 
+import { min, max, quantile } from 'd3-array';
+import { scaleLinear, scaleBand, scaleLog, scalePow } from 'd3-scale';
+import { select } from 'd3-selection';
+import { timeFormat } from 'd3-time-format';
+import Axis from '../Graph/Axis';
+import { getDateIdx, addCommas } from '../../utils/utils';
+import { margin } from '../../utils/constants';
+import { chartBkgd, green, gray, blue, scenarioColors } from '../../utils/constants'
+
+class Chart extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            severities: ['high', 'med', 'low'],
+            scaleDomains: false,
+            hoveredRect: {
+                'severity': '',
+                'scenario': '',
+                'index': 0
+            },
+            rectIsHovered: false,
+            // tooltipText: '',
+            severityMap: {
+                'high': '1% IFR, 10% hospitalization rate',
+                'med': '0.5% IFR, 5% hospitalization rate',
+                'low': '0.25% IFR, 2.5% hospitalization rate'
+            }
+            // quantileObj: {}
+        }
+        this.tooltipRef = React.createRef();
+        this.chartRef = React.createRef();
+    }
+    componentDidMount() {
+        // console.log('componentDidMount');
+        this.calculateQuantiles();
+        
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        // console.log(this.props.summaryStart, this.props.summaryEnd)
+        if (prevProps.summaryStart !== this.props.summaryStart || 
+            prevProps.summaryEnd !== this.props.summaryEnd ||
+            prevProps.dataset !== this.props.dataset ||
+            prevProps.scenarios !== this.props.scenarios ||
+            prevProps.scale !== this.props.scale) {
+            this.calculateQuantiles();
```

Comment:
You may want to log how often this gets called. Since you are trying to control the updates with this RN lifecycle (componenetDidUpdate), we want to make sure it doesn't trigger more than we want it to.

Reply:
Alternatively, you may want to go to the lifecycle event `shouldComponentUpdate` which determines if the component should even render.